solidity/random-beacon/contracts/libraries/Relay.sol

```diff
@@ -256,7 +272,33 @@ library Relay {
 
         return
             isRequestInProgress(self) &&
-            block.number > self.currentRequest.startBlock + _relayEntryTimeout;
+            block.number > self.currentRequestStartBlock + _relayEntryTimeout;
+    }
+
+    /// @notice Returns whether a relay entry request is currently in progress.
+    /// @return True if there is a request in progress. False otherwise.
+    function _isRequestInProgress(Data memory self)
+        internal
+        view
+        returns (bool)
+    {
+        return self.currentRequestID != 0;
+    }
+
+    /// @notice Returns whether the current relay request has timed out.
+    /// @return True if the request timed out. False otherwise.
+    function _hasRequestTimedOut(Data memory self)
```

Comment:
Same question about putting it next to `function hasRequestTimedOut(Data storage self)`

Reply:
If this has a very marginal gas diff, maybe we should get rid of the duplicate code?